ddtrace/_trace/processor/__init__.py

```diff
@@ -520,7 +508,7 @@ def reset(
             # Stopping them before that will raise a ServiceStatusError.
             pass
 
-        if isinstance(self.writer, AgentWriter) and appsec_enabled:
+        if isinstance(self.writer, NativeWriter) and appsec_enabled:
```

Comment:
unrelated to this PR, but it would also be nice to remove this tight coupling with appsec here